lib/cc/analyzer/engine.rb

```diff
@@ -122,7 +122,11 @@ def output_overrider
 
       # Memory limit for a running engine in bytes
       def memory_limit
-        (ENV["ENGINE_MEMORY_LIMIT_BYTES"] || DEFAULT_MEMORY_LIMIT).to_s
+        [@metadata["memory"].to_i, default_memory_limit.to_i].max.to_s
```

Comment:
I wonder if this logic is something we want to introduce right now. What do you think of adding an item in the chain of precedence but keeping out the max logic?

So the precedence would be:
- ENV
- yaml registry (new)
- `DEFAULT_MEMORY_LIMIT` constant

Reply:
@dblandin The problem with this approach is that we always have `ENGINE_MEMORY_LIMITS_BYTES` set when invoked from the `builder`. The ENV is set in `scheduler` and cascades from there, so we will never get to use the yaml entry in this scenario, and that's the reason I was treating it like a `minimum_required_memory` instead.

Reply:
Ahh, right. That makes sense. The max behavior sounds good to me then.